cclib/method/volume.py

```diff
@@ -168,10 +172,34 @@ def getbfs(ccdata):
                     bf.normalize()
                     bfs.append(bf)
 
-    #Add pyquante2 code in subsequent PR
-
+    #pyquante2
+    if _found_pyquante2:
+        for i, atom in enumerate(pymol): # `atom` is instance of pyquante2.geo.atom.atom class.
+            basis = ccdata.gbasis[i] # `basis` is basis coefficients stored in ccData.
+            for sym, primitives in basis:
+                for power in sym2powerlist[sym]: # `sym` is S, P, D, F and is used as key here.
+                    exponentlist = []
+                    coefficientlist = []
+                    
+                    for exponents, coefficients in primitives:
+                        exponentlist.append(exponents)
+                        coefficientlist.append(coefficients)
+                    
+                    basisfunction = cgbf(atom.atuple()[1:4], powers = power, exps=exponentlist, coefs=coefficientlist)
+                    basisfunction.normalize()
+                    bfs.append(basisfunction)
+                    
     return bfs
 
+# Small wrapper PyQuante & pyquante2 function that evaluates basis function on a given point
+# Used in both `wavefunction` and `electrondensity`
+def pyamp(bfs, bs, x, y, z):
+    if _found_pyquante2:

```

Comment:
I'm more convinced now that these checks should be outside the function definition for performance reasons and general consistency with what functions return (the desired value vs. `None`).

Reply:
I've added a check that raises ImportError when pyquante is not present outside the functions. Thanks.